Sparkle/SUBasicUpdateDriver.m

```diff
@@ -352,7 +352,17 @@ - (void)downloadUpdate
         request.networkServiceType = NSURLNetworkServiceTypeBackground;
     }
 
-    [request setValue:[updater userAgentString] forHTTPHeaderField:@"User-Agent"];
+    if ([updater userAgentString]) {
+        [request setValue:[updater userAgentString] forHTTPHeaderField:@"User-Agent"];
+    }
+
+    if ([updater httpHeaders]) {
+        for (NSString *key in [updater httpHeaders]) {
+            id value = [[updater httpHeaders] objectForKey:key];
```

Comment:
can you cache the values from updater to avoid repeated calls?

Reply:
Would the same need to be done in `SUAppcast.m` where the `httpHeaders` are applied?

Reply:
Do you mean like this?
```
    NSDictionary<NSString *, NSString *> *httpHeaders = [updater httpHeaders];
    if (httpHeaders) {
        for (NSString *key in httpHeaders) {
            id value = [httpHeaders objectForKey:key];
            [request setValue:value forHTTPHeaderField:key];
        }
    }
```